src/inapp/inapp.ts

```diff
@@ -349,7 +349,11 @@ export function getInAppMessages(
             );
           }
 
-          if (!payload.closeButton?.isRequiredToDismissMessage) {
+          const ua = navigator.userAgent;
+          const isSafari =
+            !!ua.match(/safari/i) && !ua.match(/chrome|chromium|crios/i);
+
+          if (!payload.closeButton?.isRequiredToDismissMessage || isSafari) {
```

Comment:
If browser is Safari, we do not render an auto-generated close button, because it requires JS to function. However, that means if the customer wants to require the close button to dismiss the message, we should override that feature by still allowing users to click outside the message to dismiss, otherwise there isn't a way to close it.